Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1495,7 +1495,13 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrackMasonry(GridTrack
         isNewEntry = true;
 
         // Correct the span as the grid item is coming from another track.
-        span = GridSpan::translatedDefiniteGridSpan(trackIndex, trackIndex + span.integerSpan());
+        unsigned shift = 0;
+        if (span.integerSpan() > 1 && ((span.integerSpan() + trackIndex) > tracks(m_direction).size())) {
```

Comment:
We could probably store `span.integerSpan() + trackIndex` in a local so we don't have to compute it twice and also potentially improve the readability

Reply:
Good idea 👍